x/axelarcork/types/params.go

```diff
@@ -0,0 +1,156 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+// Parameter keys
+var (
+	KeyEnabled         = []byte("enabled")
+	KeyIBCChannel      = []byte("ibcchannel")
+	KeyIBCPort         = []byte("ibcport")
+	KeyGMPAccount      = []byte("gmpaccount")
+	KeyExecutorAccount = []byte("executoraccount")
+	KeyTimeoutDuration = []byte("timeoutduration")
+)
+
+var _ paramtypes.ParamSet = &Params{}
+
+// ParamKeyTable returns the parameter key table.
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+// DefaultParams returns default oracle parameters
+func DefaultParams() Params {
+	return Params{
+		Enabled:         false,
+		IbcChannel:      "",
+		IbcPort:         "",
+		GmpAccount:      "",
+		ExecutorAccount: "",
+		TimeoutDuration: 0,
+	}
+}
+
+// ParamSetPairs returns the parameter set pairs.
+func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyEnabled, &p.Enabled, validateEnabled),
+		paramtypes.NewParamSetPair(KeyIBCChannel, &p.IbcChannel, validateIBCChannel),
+		paramtypes.NewParamSetPair(KeyIBCPort, &p.IbcPort, validateIBCPort),
+		paramtypes.NewParamSetPair(KeyGMPAccount, &p.GmpAccount, validateGMPAccount),
+		paramtypes.NewParamSetPair(KeyExecutorAccount, &p.ExecutorAccount, validateExecutorAccount),
+		paramtypes.NewParamSetPair(KeyTimeoutDuration, &p.TimeoutDuration, validateTimeoutDuration),
+	}
+}
+
+// ValidateBasic performs basic validation on oracle parameters.
+func (p *Params) ValidateBasic() error {
+	if err := validateEnabled(p.Enabled); err != nil {
+		return err
+	}
+
+	if p.Enabled {
+		if err := validateIBCChannel(p.IbcChannel); err != nil {
+			return err
+		}
+		if err := validateIBCPort(p.IbcPort); err != nil {
+			return err
+		}
+		if err := validateGMPAccount(p.GmpAccount); err != nil {
+			return err
+		}
+		if err := validateExecutorAccount(p.ExecutorAccount); err != nil {
+			return err
+		}
+		if err := validateTimeoutDuration(p.TimeoutDuration); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateEnabled(i interface{}) error {
+	_, ok := i.(bool)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	return nil
+}
+
+func validateIBCChannel(i interface{}) error {
+	ibcChannel, ok := i.(string)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if ibcChannel == "" {
```

Comment:
I wonder if the IBC module has any validations around channel/port that we could import or use -- there should be expected formats for these strings.

Reply:
Yes, found and used them from ibc-go